### PR TITLE
Nav and NavLink - updated rails docs

### PR DESCRIPTION
**Pull request by QuintonJason**
## Description

- [x] - update demo to use rails code
- [x] - a11y added prop to include `aria-label`

## Test notes

No visual changes

### Steps for testing
1. Visit the sage rails nav page: http://localhost:4000/pages/object/nav
2. Verify the updates to docs and demo
3. Visit the sage rails nav link page: http://localhost:4000/pages/element/nav_link

### CODE REVIEW

**Comment by QuintonJason** [comment]
Not ready for review. Sorry for early request

**Comment by QuintonJason** [comment]
Screen Shot 2021-01-20 at 9 53 13 AM

One issue I need some assistance pushing through is the `aria-label` not escaping properly. Any ideas?

File: docs/app/views/examples/elements/nav_link/_props.html.erb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<tr>
+  <td>icon</td>
+  <td>Allows for configurations for an and its placement to be provided.</td>
+  <td>String</td>
+  <td>null</td>
+</tr>
+<tr>
+  <td>link</td>
+  <td>This sets the `href` for the link.</td>
```
Comment (philschanely) [inline docs/app/views/examples/elements/nav_link/_props.html.erb:9]: Consider adding the `md()` function here to render this as markdown:

```erb
<td><%= md("This sets the `href` for the link.") %></td>
```

File: docs/app/views/examples/objects/nav/_preview.html.erb [hunk 2]
```diff
@@ -1,46 +1,44 @@
...
-      text: "Nav Link 4",
-      is_active: false,
-      url: "#"
-    },
-    {
-      text: "Nav Link 5",
-      is_active: false,
-      url: "#"
+<%= sage_component SageNav, {
+  ariaLabel: "My Aria Label",
```
Comment (philschanely) [inline docs/app/views/examples/objects/nav/_preview.html.erb:2]: Testing locally this needed to be `aria_label`.

**Review by philschanely — APPROVED** [review]
A few small tweaks and :thumbs:

File: docs/lib/sage_rails/app/views/sage_components/_sage_nav.html.erb [hunk 3]
```diff
@@ -1,3 +1,3 @@
-<div class="sage-nav">
+<div class="sage-nav"  <%= "aria-label=\"#{component.aria_label}\"" if component.aria_label.present? %> ">
```
Comment (monkeypox8) [inline docs/lib/sage_rails/app/views/sage_components/_sage_nav.html.erb:1]: As you noted, the quote marks don't appear to be escaping as expected once rendered. There's also a stray `"` at the end of the tag:

Screen Shot 2021-01-21 at 5 44 38 PM

I tried a few different variations, but none of them seemed to be responding to the `\` escape character, not sure why. For now, I'd suggest using the less condensed version instead to circumvent the issue:
```<div class="sage-nav"<% if component.aria_label.present? %> aria-label="<%= component.aria_label %>"<% end %>>```

**Review by monkeypox8 — COMMENTED** [review]
Looking great! I noted something strange happening with the `\` escape character when rendering the `aria` output.

**Pull request merged**